### PR TITLE
Added util func for skymapper weather and query/store in db script

### PR DESCRIPTION
**Pull request by fergusL**
added a util that will return a skymapper weather data dict of form,

```
{'Date': '2021-10-08',
 'Time': '01:53:00',
 'Raining': False,
 'Cloudy': True,
 'Humidity': False,
 'Windy': False,
 'Hailing': False,
 'Excessive_Light': True,
 'Internal_Rain_Sensor': False,
 'External_Rain_Sensor': False,
 'Relative_Humidity': '48',
 'Wind_Speed_Minimum': '3.60',
 'Wind_Speed_Average': '4.06',
 'Wind_Speed_Maximum': '4.20',
 'Sky_Minus_Ambient': '-32.00',
 'Wind_Speed': '10.40'}
```

and a script that can be run to add this data to weather db periodically

### CODE REVIEW

**Comment by fergusL** [comment]
- cleaned up functions that fetch aat/skymapper weather data
- added function/config for determining which sources to use and which parameters to base safety decisions on
- added generic script for logging alternate weather data
- updated huntsman.pocs.core to use alternate weather sources as specified in config (ie whether to use a source and what weather readings to use)
- updated testing config to ignore alt weather sources

File: scripts/read_alt_weather.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+
+def main(source,
+         config,
+         store_result,
+         storage_dir,
+         read_delay,
+         verbose,
+         **kwargs):
+
```
Comment (danjampro) [inline scripts/read_alt_weather.py:20]: Could add a docstring here explaining what the function does and what the args are.

File: scripts/read_alt_weather.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            config = yaml.safe_load(f)
+            alt_weather_config = config.get('alt_weather_sources').get(source)
+
+    if alt_weather_config is None:
+        raise ValueError("The weather source config is None, please supply a config.")
+
+    while True:
+        try:
+            # load and call function for fetching alterant weather source data
+            data = load_module('huntsman.pocs.utils.weather.get_' + source)()
```
Comment (danjampro) [inline scripts/read_alt_weather.py]: Could take the `load_module` call outside of the loop

File: scripts/read_alt_weather.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            # determine if the weather reading indicates safe conditions
+            data = determine_alt_weather_safety(data, alt_weather_config)
+            if verbose:
+                print(f'{data!r}')
```
Comment (danjampro) [inline scripts/read_alt_weather.py]: Can we use the logger instead?

File: src/huntsman/pocs/core.py [hunk 4]
```diff
@@ -114,21 +112,51 @@ def is_weather_safe(self, **kwargs):
         if self._in_simulator('weather'):
             return True
         # if not in simulator mode, determine safety from huntsman weather data
-        is_safe = super().is_weather_safe(**kwargs)
+        is_safe_list = [super().is_weather_safe(**kwargs)]
+
+        # check alternate sources
+        alt_weather_config = self.get_config("alt_weather_sources", default=None)
+        sources_to_check = [s for s in alt_weather_config.keys()
```
Comment (danjampro) [inline src/huntsman/pocs/core.py]: Think we can simplify this:
`[k for k, v in alt_weather_config.items() if v['use']]`

**Review by danjampro — APPROVED** [review]
LGTM! A few minor comments above.

**Pull request merged**